src/garrysmod/gamemodes/zombiesurvival/gamemode/sv_database.lua

```diff
@@ -0,0 +1,32 @@
+require "pg"
+
+local database = CreateConVar("postgres_db", "zombiesurvival", bit.bor(FCVAR_PROTECTED), "")
+local password = CreateConVar("postgres_password", "zombiesurvival", bit.bor(FCVAR_PROTECTED), "")
+local user     = CreateConVar("postgres_user", "zombiesurvival", bit.bor(FCVAR_PROTECTED), "")
+local host     = CreateConVar("postgres_host", "database", bit.bor(FCVAR_PROTECTED), "")
+
+GM.Database = GM.Database or {}
+
+function GM.Database:EstablishConnection(onConnected)
+    local host, user, password, port, database = host:GetString(), user:GetString(), password:GetString(), database:GetString()
+
+    if pg then
+        self.conn = pg.new_connection()
+
+        local success, err = self.conn:connect(host, user, password, database, 5432)
+
+        if success then
+            success, err = self.conn:set_encoding("UTF8")
+
+            if not success then
+                ErrorNoHalt("Failed to set connection encoding:\n", err)
+            end
+
+            if isfunction(onConnected) then onConnected(self) end
+        else
+            ErrorNoHalt("Unable to connect to the database!\n", err)
+        end
+    else
+        ErrorNoHalt("gmsv_pg has not been found.")
+    end
+end
```

Comment:
A lot of this is [borrowed from](https://github.com/TeslaCloud/flux-ce/blob/4ff9bb15e6063b85e49099210863144677baec6b/crates/active_record/lib/adapters/pg.lua#L29) the Active Record ORM built by gmsv_pg's maintainer.

Reply:
Function is pretty self-explanatory, but is there any consideration needed for usage of comments/docblock/function definitions

Reply:
Thats fine, but I wouldn't use 'existing code is bad' as a reason against doing so.